src/CodeGenerator/AutoloadGenerator.php

```diff
@@ -49,6 +49,7 @@ private function buildFromTemplate(string $templateFile, string $outputFile, arr
         $template = file_get_contents($templateFile);
 
         assert(is_string($template));
+        assert(is_string($this->outputDirectory));
```

Comment:
Imho, thats not safe. This is only true in case someone called the setter. Using php `assert` in this case is *not* how it should be used.

https://github.com/laminas/technical-steering-committee/blob/main/meetings/minutes/2021-08-02-TSC-Minutes.md#allow-php-assert-along-with-webmozart-assert-in-specific-cases

Reply:
You are right. `$this->ensureOutputDirectory()` would be correct here, which does provide the correct assert and the required directory existence checks / create attempts.